### PR TITLE
Add GLU activation

### PR DESCRIPTION
**Pull request by shashaka**
I noticed that the GLU activation function is not currently supported in Keras, despite its availability in the Torch and JAX frameworks. To address this gap, I implemented the GLU logic for both TensorFlow and NumPy, leveraging the existing implementations in JAX and Torch for consistency.

If this approach is acceptable, I can proceed by adding test cases to ensure the correctness of these implementations.

### CODE REVIEW

File: keras/src/backend/numpy/nn.py [hunk 1]
```diff
@@ -121,6 +121,13 @@ def celu(x, alpha=1.0):
     )
 
 
+def glu(x, axis=-1):
+    x = convert_to_tensor(x)
+    assert x.shape[axis] % 2 == 0, "axis size must be divisible by 2"
```
Comment (fchollet) [inline keras/src/backend/numpy/nn.py:126]: Use a `ValueError` and specify "Received: x.shape={x.shape} with axis={axis}" in the error message

File: keras/src/backend/tensorflow/nn.py [hunk 2]
```diff
@@ -82,6 +82,12 @@ def celu(x, alpha=1.0):
     )
 
 
+def glu(x, axis=-1):
+    assert x.shape[axis] % 2 == 0, "axis size must be divisible by 2"
```
Comment (fchollet) [inline keras/src/backend/tensorflow/nn.py:86]: Same here

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! We can add it to the API.

File: keras/src/backend/tensorflow/nn.py (hunk 2 above)
Comment (shashaka) [reply keras/src/backend/tensorflow/nn.py]: I also updated this.

**Comment by shashaka** [comment]
@fchollet 
I've updated the code based on your guidance. Please let me know if there are any further changes or adjustments you'd recommend.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
